src/azure-cli/azure/cli/command_modules/mysql/custom.py

```diff
@@ -58,6 +58,111 @@ def server_list_custom_func(client, resource_group_name=None):
     return client.list()
 
 
+def flexible_server_threat_model_update(cmd, client, resource_group_name,
+                                        server_name,
+                                        defender_state,
+                                        advanced_threat_protection_name,
+                                        subscription_id=None
+                                        ):
+
+    allowed_defender_name_values = [mysql_flexibleservers.models.AdvancedThreatProtectionName.DEFAULT.value]
+    if advanced_threat_protection_name not in allowed_defender_name_values:
+        raise ValueError("Invalid defender protection name provided, Allowed value - {}".format(allowed_defender_name_values))
+
+    allowed_defender_state_values = [mysql_flexibleservers.models.AdvancedThreatProtectionState.ENABLED.value, mysql_flexibleservers.models.AdvancedThreatProtectionState.DISABLED.value]
+    if defender_state not in allowed_defender_state_values:
+        raise ValueError("Invalid defender state provided , Allowed value - {}".format(allowed_defender_state_values))
+
+    # parameters = {
+    #     'state': defender_state
+    # }
+    response = {
+        "id": "/subscriptions/00000000-1111-2222-3333-444444444444/resourceGroups/threatprotection-4799/providers/Microsoft.DBforMySQL/flexibleServers/threatprotection-6440/advancedThreatProtectionSettings/Default",
+        "name": "Default",
+        "type": "Microsoft.DBforMySQL/flexibleServers/advancedThreatProtectionSettings",
+        "systemData": {
+            "createdBy": "string",
+            "createdByType": "User",
+            "createdAt": "2022-04-03T04:41:33.937Z",
+            "lastModifiedBy": "string",
+            "lastModifiedByType": "User",
+            "lastModifiedAt": "2022-04-03T04:41:33.937Z"
+        },
+        "properties": {
+            "state": "{fname}",
+            "creationTime": "2022-04-03T04:41:33.937Z",
+            "provisioningState": "Succeeded"
+        }
+    }
+    response["properties"]["state"] = response["properties"]["state"].format(fname=defender_state)
+    return response
```

Comment:
Is this mock code? You are just returning hard coded response here. Even if it is mock code, you can make a call to the sdk method here and return response instead of hard coding in the code. Expected response can be added in comment as an example.

Reply:
Done, intially "No resource provider found " were the only response to any changes in the sdk so had to hardcode in the command itself, but now its fixed.